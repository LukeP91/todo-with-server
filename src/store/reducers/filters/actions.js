import { types } from './types'

export const setFilter = value => {
  return {
    type: types.SET_FILTER,
    payload: value,
  }
}

export const setVisiblityFilter = value => {
  return {
    type: types.SET_VISIBILITY_FILTER,
    payload: value,
  }
}
