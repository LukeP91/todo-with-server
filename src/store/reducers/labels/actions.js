import { types } from './types'

const fetchLabelsRequest = () => {
  return {
    type: types.FETCH_LABELS_REQUEST,
  }
}

const fetchLabelsSuccess = payload => {
  return {
    type: types.FETCH_LABELS_SUCCESS,
    payload,
  }
}

const fetchLabelsError = () => {
  return {
    type: types.FETCH_LABELS_ERROR,
  }
}

const fetchData = (url, params = {}) => {
  return fetch(url, params).then(response => Promise.all([response, response.json()]))
}

export const fetchLabels = (url, params) => {
  return dispatch => {
    dispatch(fetchLabelsRequest())
    return fetchData(url, params).then(([response, json]) => {
      if (response.status === 200) {
        dispatch(fetchLabelsSuccess(json))
      } else {
        dispatch(fetchLabelsError())
      }
    })
  }
}
