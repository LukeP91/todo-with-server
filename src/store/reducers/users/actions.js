import { types } from './types'

const fetchUsersRequest = () => {
  return {
    type: types.FETCH_USERS_REQUEST,
  }
}

const fetchUsersSuccess = payload => {
  return {
    type: types.FETCH_USERS_SUCCESS,
    payload,
  }
}

const fetchUsersError = () => {
  return {
    type: types.FETCH_USERS_ERROR,
  }
}

const fetchData = (url, params = {}) => {
  return fetch(url, params).then(response => Promise.all([response, response.json()]))
}

export const fetchUsers = (url, params) => {
  return dispatch => {
    dispatch(fetchUsersRequest())
    return fetchData(url, params).then(([response, json]) => {
      if (response.status === 200) {
        dispatch(fetchUsersSuccess(json))
      } else {
        dispatch(fetchUsersError())
      }
    })
  }
}
