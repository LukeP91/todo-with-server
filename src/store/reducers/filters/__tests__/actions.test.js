import fetchMock from 'fetch-mock'

import * as actions from '../actions'

describe('actionCreators', () => {
  afterEach(() => {
    fetchMock.reset()
    fetchMock.restore()
  })

  it('creates action to set filter', () => {
    const text = 'test'
    expect(actions.setFilter(text)).toMatchSnapshot()
  })

  it('creates action to set visiblity filter', () => {
    const text = 'ALL'
    expect(actions.setVisiblityFilter(text)).toMatchSnapshot()
  })
})
