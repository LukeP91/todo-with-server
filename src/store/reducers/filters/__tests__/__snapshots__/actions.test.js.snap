// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionCreators creates action to set filter 1`] = `
Object {
  "payload": "test",
  "type": "SET_FILTER",
}
`;

exports[`actionCreators creates action to set visiblity filter 1`] = `
Object {
  "payload": "ALL",
  "type": "SET_VISIBILITY_FILTER",
}
`;
